pandas/tests/tslibs/test_timedeltas.py

```diff
@@ -34,3 +34,41 @@ def test_huge_nanoseconds_overflow():
     # GH 32402
     assert delta_to_nanoseconds(Timedelta(1e10)) == 1e10
     assert delta_to_nanoseconds(Timedelta(nanoseconds=1e10)) == 1e10
+
+
+class TestComputingDeltasWithTimezoneDifferences:
+    def test_fails_to_subtract_when_one_date_is_naive(self):
+        time1 = Timestamp("2020-10-28 22:08:00-04:00")
+        time2 = Timestamp("2020-10-28 22:08:00")
+
+        msg = (
+            "Cannot compute time difference between "
+            "timezone-aware and naive datetime values."
+        )
+        with pytest.raises(TypeError, match=msg):
+            time1 - time2
+
+        with pytest.raises(TypeError, match=msg):
+            time2 - time1
+
+    def test_subtracts_dates_of_different_timezones(self):
+        time1 = Timestamp("2020-10-28 22:08:00-04:00")
+        time2 = Timestamp("2020-10-28 22:08:00-00:00")
```

Comment:
i'm having a hard time seeing what the can of worms will be here, can you elaborate?

to me, the complex part is determining the timezone if it is specified using a different library (e.g. `pytz`, `dateutil` and `datetime` all have different `implementations`). 

the actual subtraction part is just, you add the utc offset to the number of hours difference. pandas can already do it, and it would match the implementation of `datetime` which does support this operation.


Are there historical examples you can point to where this has been a significant problem?

Reply:
@jreback the simplest reason to allow this is because the stdlib datetime does

@Anjoman the the tzinfo library pytz/dateutil/timezone is irrelevant.  once you know that `not ((self.tzinfo is None) ^ (other.tzinfo is None))`, the difference is `Timedelta(self.value - other.value)` regardless.